core/adapters/random.go

```diff
@@ -56,33 +69,75 @@ func (ra *Random) TaskType() models.TaskType {
 
 // Perform returns the the proof for the VRF output given seed, or an error.
 func (ra *Random) Perform(input models.RunInput, store *store.Store) models.RunOutput {
-	key, err := getKey(ra, input)
+	key, preSeed, blockHash, blockNum, err := getInputs(ra, input, store)
 	if err != nil {
-		return models.NewRunOutputError(errors.Wrapf(err, "bad key for vrf task"))
-	}
-	seed, err := getSeed(input)
-	if err != nil {
-		return models.NewRunOutputError(errors.Wrap(err, "bad seed for vrf task"))
+		return models.NewRunOutputError(err)
 	}
-	solidityProof, err := store.VRFKeyStore.GenerateProof(key, seed)
+	solidityProof, err := store.VRFKeyStore.GenerateProof(key, preSeed,
+		*blockHash, *blockNum)
 	if err != nil {
 		return models.NewRunOutputError(err)
 	}
 	ethereumByteArray := fmt.Sprintf("0x%x", utils.EVMEncodeBytes(solidityProof[:]))
 	return models.NewRunOutputCompleteWithResult(ethereumByteArray)
 }
 
-// getSeed returns the numeric seed for the vrf task, or an error
-func getSeed(input models.RunInput) (*big.Int, error) {
+// getInputs parses the JSON input and returns an error, or the values in it.
+func getInputs(ra *Random, input models.RunInput, store *store.Store) (
+	key *vrfkey.PublicKey, preSeed *vrf.Seed, blockHash *common.Hash,
+	blockNum *uint64, err error) {
+	key, err = getKey(ra, input)
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrapf(err, "bad key for vrf task")
+	}
+	preSeed, err = getPreSeed(input)
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrap(err, "bad seed for vrf task")
+	}
+	blockHash, blockNum, err = getBlockData(input)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+	return key, preSeed, blockHash, blockNum, nil
+}
+
+// getBlockData parses the block-related data from the JSON input
+func getBlockData(input models.RunInput) (
+	blockHash *common.Hash, blockNum *uint64, err error) {
+	hashBytes, err := extractHex(input, "blockHash")
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "bad blockHash for vrf task")
+	}
+	bHash := common.BytesToHash(hashBytes)
+
+	rawBlockNum := input.Data().Get("blockNum")
+	if rawBlockNum.Type != gjson.Number {
+		return nil, nil, errors.Errorf("blockNum field has no number: %+v",
```

Comment:
This wrapping is weird... why not just let it all on one line?

Reply:
I prefer to keep textwidth under 80 columns, so I can view large amounts of related code in three parallel columns on the one screen. E.g.
![Screenshot from 2020-05-29 11-11-37](https://user-images.githubusercontent.com/70152/83275625-62569400-a19d-11ea-8a4e-33602505c0d0.png)

Reply:
It's a shame the formatter doesn't have an opinion on this... but since it doesn't, I will refrain from bikeshedding. It doesn't matter too much tbh.